### PR TITLE
Use array destructuring for some simple case

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/document/doc-printer.js [hunk 1]
```diff
@@ -154,9 +154,7 @@ function fits(next, restCommands, width, options, mustBeFlat) {
     }
 
     const x = cmds.pop();
-    const ind = x[0];
-    const mode = x[1];
-    const doc = x[2];
+    const [ind, mode, doc] = x;
```
Comment (jamesgeorge007) [inline src/document/doc-printer.js]:
```suggestion
    const [ind, mode, doc] = cmds.pop();
```

Comment (jamesgeorge007) [reply src/document/doc-printer.js]: I can't see the variable `x` serving it's purpose anywhere else.

File: src/document/doc-printer.js [hunk 2]
```diff
@@ -254,9 +252,7 @@ function printDocToString(doc, options) {
 
   while (cmds.length !== 0) {
     const x = cmds.pop();
-    const ind = x[0];
-    const mode = x[1];
-    const doc = x[2];
+    const [ind, mode, doc] = x;
```
Comment (jamesgeorge007) [inline src/document/doc-printer.js]:
```suggestion
    const [ind, mode, doc] = cmds.pop();
```

File: src/document/doc-printer.js [hunk 3]
```diff
@@ -254,9 +252,7 @@ function printDocToString(doc, options) {
 
   while (cmds.length !== 0) {
     const x = cmds.pop();
```
Comment (jamesgeorge007) [inline src/document/doc-printer.js]:
```suggestion
```

File: src/document/doc-printer.js [hunk 4]
```diff
@@ -154,9 +154,7 @@ function fits(next, restCommands, width, options, mustBeFlat) {
     }
 
     const x = cmds.pop();
```
Comment (jamesgeorge007) [inline src/document/doc-printer.js]:
```suggestion
```

**Comment by thorn0** [comment]
I'm really not sure the readability improves in simple cases like this (with only one variable declared):
```diff
-  const child = node.children[0];
+  const [child] = node.children;
```
Let's keep them as is.

File: src/document/doc-printer.js (hunk 1 above)
Comment (fisker) [reply src/document/doc-printer.js]: Good idea, updated. Thanks!

**Pull request merged**